auditlog/audit_log.go

```diff
@@ -0,0 +1,227 @@
+package auditlog
+
+import (
+	"context"
+	"database/sql"
+	"database/sql/driver"
+	"encoding/json"
+	"reflect"
+	"time"
+
+	"github.com/fabric8-services/fabric8-common/closeable"
+	"github.com/fabric8-services/fabric8-common/convert"
+	"github.com/fabric8-services/fabric8-common/errors"
+	"github.com/fabric8-services/fabric8-common/log"
+
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	uuid "github.com/satori/go.uuid"
+)
+
+// AuditLog an audit log record to track usage of the service
+type AuditLog struct {
+	ID          uuid.UUID   `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"`
+	CreatedAt   time.Time   `json:"created_at,omitempty"`
```

Comment:
Having `deleted_at` leaves a provision for soft deletes. Any reason why we are skipping that?

Reply:
I was thinking along the same lines as well :) Just that, adding support for soft delete helps us keep the data forever and also hide it *if needed* .

( note: this is not a blocking review comment )